gateway/implementation/src/main/java/eu/openg/guessnumberapi/gateway/implementation/PostgreSQL/PostgresqlGameRepo.java

```diff
@@ -0,0 +1,110 @@
+package eu.openg.guessnumberapi.gateway.implementation.PostgreSQL;
+
+import eu.openg.guessnumberapi.domain.Game;
+import eu.openg.guessnumberapi.gateway.api.GameRepository;
+import eu.openg.guessnumberapi.gateway.implementation.exception.PostgresqlException;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+
+public class PostgresqlGameRepo implements GameRepository {
+    private static final Logger LOGGER = LogManager.getLogger(PostgresqlGameRepo.class);
+    private final PostgresqlConnection postgresqlConnection;
+
+    public PostgresqlGameRepo(PostgresqlConnection postgresqlConnection) {
+        this.postgresqlConnection = postgresqlConnection;
+        createGameTableIfNotExists();
+    }
+
+    private void createGameTableIfNotExists() {
+        try (Statement statement = postgresqlConnection.getConnection().createStatement()) {
+            statement.executeUpdate(QueryUtils.CREATE_TABLE_QUERY);
+            LOGGER.info("PostgreSql database table {} created successfully", QueryUtils.GAME_TABLE_NAME);
+        } catch (SQLException e) {
+            throw logErrorAndReturnNewException("CREATE TABLE failed. game table could not be created.", e);
+        }
+    }
+
+    @Override
+    public int saveNewGameAndReturnId(Game game) {
+        try (PreparedStatement statement = prepareInsertQuery(game);
+             ResultSet resultSet = statement.executeQuery()) {
+            int id = extractFieldFromResultSet(resultSet, QueryUtils.GAME_ID, game.getGameId());
+            LOGGER.info("PostgreSql new game with gameID: {} insert successful.", id);
+            return id;
+        } catch (SQLException e) {
+            throw logErrorAndReturnNewException("INSERT failed. New game could not be inserted.", e);
+        }
+    }
+
+    private PreparedStatement prepareInsertQuery(Game game) throws SQLException {
```

Comment:
čia dar galima padaryti createStatement(String query, int... params) ir td galėsi reusint šitą metodą visais case'ais, nepriklausomai nuo paramų skaičiaus